stock_indicators/indicators/fractal.py

```diff
@@ -1,23 +1,20 @@
 from decimal import Decimal
-from typing import Iterable, Optional, TypeVar
+from typing import Iterable, Optional, TypeVar, overload
 
 from stock_indicators._cslib import CsIndicator
 from stock_indicators._cstypes import List as CsList
 from stock_indicators._cstypes import Decimal as CsDecimal
 from stock_indicators._cstypes import to_pydecimal
+from stock_indicators.indicators.common.enums import EndType
 from stock_indicators.indicators.common.results import IndicatorResults, ResultBase
 from stock_indicators.indicators.common.quote import Quote
 
-#TODO: Needs to support EndType enums.
-# @overload
-# def get_fractal(quotes: Iterable[Quote], window_span: int = 2) -> "FractalResults": ...
-# @overload
-# def get_fractal(quotes: Iterable[Quote], left_span: int, right_span: int) -> "FractalResults": ...
-# def get_fractal(quotes: Iterable[Quote], left_span: int, right_span: int = None):
-#     fractal_results = CsIndicator.GetFractal[Quote](CsList(Quote, quotes), left_span, right_span)
-#     return FractalResults(fractal_results, FractalResult)
 
-def get_fractal(quotes: Iterable[Quote], window_span: int = 2):
+@overload
+def get_fractal(quotes: Iterable[Quote], window_span: int = 2, end_type = EndType.HIGH_LOW) -> "FractalResults[FractalResult]": ...
+@overload
+def get_fractal(quotes: Iterable[Quote], left_span: int, right_span: int, end_type = EndType.HIGH_LOW) -> "FractalResults[FractalResult]": ...
+def get_fractal(quotes, left_span = None, right_span = EndType.HIGH_LOW, end_type = EndType.HIGH_LOW):
```

Comment:
`right_span` value here is not an `enum`

Reply:
Thanks, it was enum problem.

Reply:
I was looking at `right_span = EndType.HIGH_LOW` specifically.  right + left span are user-supplied integer values and would never be an `EndType` enum value.

Reply:
Ah, right. but it's just for the role as a third parameter of the first overloaded method.